packages/@react-spectrum/dialog/docs/Dialog.mdx

```diff
@@ -37,58 +46,371 @@ import {Heading, Text} from '@react-spectrum/typography';
+of the Dialog's action buttons. The following example alerts if the Dialog's save or cancel button is clicked.
+
+```tsx example
+function Example() {
+  let alertSave = (cb) => {
+    cb();
+    alert('Profile saved!');
+  }
+
+  let alertCancel = (cb) => {
+    cb();
+    alert('Profile not saved!');
+  }
 
-## Validation
+  return (
+    <DialogTrigger>
+      <ActionButton>Set Profile</ActionButton>
+      {(close) => (
+        <Dialog>
+          <Heading>Profile</Heading>
+          <Divider />
+          <ButtonGroup>
+            <Button autoFocus variant="cta" onPress={() => alertSave(close)}>Save</Button>
+            <Button variant="secondary" onPress={() => alertCancel(close)}>Cancel</Button>
+          </ButtonGroup>
+          <Content>
+            <Form>
+              <TextField label="Name" />
+              <Checkbox>Make private</Checkbox>
+            </Form>
+          </Content>    
+        </Dialog>
+      )}
+    </DialogTrigger>
+  );
+}
+```
+
+Additionally, DialogTrigger accepts an `onOpenChange` prop which is triggered whenever the Dialog is opened or closed. 
+For more information on the `onOpenChange` prop see [DialogTrigger page](TODO LINK HERE).
+
+### Dismissable Dialogs
+
+Dismissable Dialogs support a `onDismiss` prop which is triggered whenever the Dialog's close button is clicked.
+Similar to non-dismissable dialogs, you must chain the DialogTrigger's `close` function with whatever callback you provide as
+`onDismiss`.
 
-*If the component supports validation props, include this section.*
+```tsx example
+function Example() {
+  let alertDismiss = (cb) => {
+    cb();
+    alert('Dialog dismissed.');
+  }
+
+  return (
+    <DialogTrigger isDismissable>
+      <ActionButton>Info</ActionButton>
+      {(close) => (
+        <Dialog onDismiss={() => alertDismiss(close)}>
+          <Heading>Version Info</Heading>
+          <Divider />
+          <Content>
+            <Text>
+              Version 1.0.0, Copyright 2020
+            </Text>
+          </Content>    
+        </Dialog>
+      )}
+    </DialogTrigger>
+  );
+}
+```
+
+Note that `onDismiss` is optional. If you don't need to add a `onDismiss` handler to your dismissable Dialog, you may omit
+the wrapping `close` function surrounding the Dialog. An example of this can be found in the Examples section [above](#examples).
 
 ## Props
 
 <PropTable component={docs.exports.Dialog} links={docs.links} />
 
 ## Visual Options
 
-*Show examples of all variants and visual props here with links to the design website for more usage details. Examples can be grouped together for conciseness.*
+### Dialog Variants
+
+Dialogs can be rendered as modals, popovers, or trays. See the [DialogTrigger docs](TODO link) for more information.
+
+```tsx example
+<DialogTrigger type="modal">
+  <ActionButton>Trigger Modal</ActionButton>
+  <Dialog>
+    <Heading>Modal</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        This is a modal.
```

Comment:
Or add to this text "Press 'Esc' key to close."

Reply:
gonna add isDismissable